File: Beginner/Sums in a Triangle  (SUMTRIAN)/triangle.py
t = int(input())

for i in range(t):
    n = int(input())
    arr = []
    for x in range(n):
        a = [int(x) for x in input().split()] 
        arr.append(a)
    
    for x in range(n-2, -1, -1):
        for y in range(0, x+1):
            arr[x][y] = arr[x][y] + max(arr[x+1][y], arr[x+1][y+1])
    print(arr[0][0])

File: Beginner/Sums in a Triangle  (SUMTRIAN)/triangle.java
import java.util.*;
import java.math.*;

class Codechef {
	public static void main (String[] args) throws java.lang.Exception {
		Scanner scan = new Scanner(System.in);
		int t = scan.nextInt();
		while(t--!=0) {
            int n = scan.nextInt();
            int arr[][] = new int[n+1][n+1];
            for(int i=1; i<=n; i++) {
                for(int j=1; j<=i; j++)
                    arr[i][j] = scan.nextInt();
            }
		
		for(int i=n-1; i>=1; i--) {
		    for(int j=1; j<=i; j++) {
		        arr[i][j] = arr[i][j]+ Math.max(arr[i+1][j], arr[i+1][j+1]);
		    }
		}
		System.out.println(arr[1][1]);
        }
        scan.close();
	}
}